src/components/ToggleButton/ToggleButton.js

```diff
@@ -1,134 +1,164 @@
 import classNames from 'classnames';
 import PropTypes from 'prop-types';
-import React, { Component } from 'react';
+import React from 'react';
 import { Button } from 'react-bootstrap';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 
 export const toggleButtonProps = {
   variants: ['check', 'circle', 'switch', 'grid-list']
 };
 
-export default class ToggleButton extends Component {
-  static propTypes = {
-    className: PropTypes.string,
-    buttonProps: PropTypes.object,
-    iconProps: PropTypes.object,
-    title: PropTypes.string.isRequired,
-    value: PropTypes.bool,
-    iconOnly: PropTypes.bool,
-    variant: PropTypes.oneOf(toggleButtonProps.variants),
-    onClick: PropTypes.func
-  };
+function Switch(props) {
+  const { className, value, title, onClick } = props;
 
-  static defaultProps = {
-    buttonProps: {},
-    iconProps: {},
-    value: false,
-    variant: 'check'
-  };
+  return (
+    <label className="switch my-auto mx-1">
+      <input type="checkbox" checked={value} onChange={onClick} name={title} />
+      <span className={classNames(className, 'slider slider--round')}></span>
+    </label>
+  );
+}
 
-  constructor(props) {
-    super(props);
+Switch.propTypes = {
+  className: PropTypes.string,
```

Comment:
There's a lot of repeated propTypes. I was thinking of moving all of them from the ToggleButton into a `const propTypes` and `const defaultProps` at the top.

What's your opinion @ayan4m1

Reply:
Actually, I kind of like this approach better:
```
IconOnly.propTypes = Switch.propTypes = GridList.propTypes = ButtonElement.propTypes = ToggleButton.propTypes = {
  className: PropTypes.string,
  buttonProps: PropTypes.object,
  iconProps: PropTypes.object,
  title: PropTypes.string.isRequired,
  value: PropTypes.bool,
  iconOnly: PropTypes.bool,
  variant: PropTypes.oneOf(toggleButtonProps.variants),
  onClick: PropTypes.func
};

IconOnly.defaultProps = Switch.defaultProps = GridList.defaultProps = ButtonElement.defaultProps = ToggleButton.defaultProps = {
  buttonProps: {},
  iconProps: {},
  value: false,
  variant: 'check'
};
```

Reply:
made the change in a24b31bdcb5545ebaf624f97fa752ff48086cbf2